### ISSUE TITLE
[BUG] Sigma Color 400 with some 8088 XT machines

### ISSUE THREAD

**Issue by Fenix770** [86Box/86Box#338]
I tested the card with Windows 1.04 and the 4 and 8 color drivers.  Works with AMI 286 machine andgeneric XT clone. But does not work with AMI XT and Phoenix XT.  (I did not tested others machines yet) 

The AMI gives an stack error in MS-DOS 3.20 and Phoenix does not even count the memory at boot time.

Here is a screenshoot.

![20181104_212306](https://user-images.githubusercontent.com/31600563/47972221-e3f54700-e078-11e8-9a05-d5d4e385fe37.png)

Thanks for this new graphics card!

**Comment by OBattler** [comment]
Could you upload the hard disk image, configuration file, and NVR, please, so I can reproduce this bug and see what's wrong?

**Comment by Fenix770** [comment]
Hello.
The hard disk contains a common XT configuration with 640kb RAM Win 1.04. DOS 3.2 and Sigma 400 using the 8 colors driver. (I don't know if Win 2.x divers exist)

The old bug does not apper anymore in the 86Box-32-a3fb51ac commit.

However, depending on the XT machine selected, When I do a dir.exe inside DOS directory, 
it may:
1. Lock up the system or
2. The cursor list all files but it dispears out of the secreen or
3. It simply does not clear the screen but only the last line

Here is the Hard disk image file:

[Sigma color 400 Win 2.zip](https://github.com/86Box/86Box/files/3209973/Sigma.color.400.Win.2.zip)

**Comment by OBattler** [comment]
Would it be possible to have 86box.cfg? Because I don't even know the CHS parameters of the hard disk image, for example.

**Comment by OBattler** [comment]
OK, I got it to work (turns out the CHS was a standard 615/4/17). On the AMI BIOS, the system locks up.

**Comment by OBattler** [comment]
I'm going to mark this as "not sure it is an emulator bug". It might be that some clone BIOS'es were not fully compatible with the Sigma.

**Comment by OBattler** [comment]
OK, I get the same problem on IBM XT 1986. The problem goes away if I switch to the CGA.

**Comment by OBattler** [comment]
Try IBM PC DOS 3.20, see if the problem also exists there.

**Comment by OBattler** [comment]
Of the XT machine, the emulated Sigma seems to work on: Generic XT Clone, OpenXT, and the DTK XT clone.

**Comment by Fenix770** [comment]
I have done some testing now. I found two issues with the Color 400.

One is the some 286 BIOS complains as wrong configured if I select CGA as a graphics option. This is the case of IBM AT with AMI BIOS says "video  mode mismatch" and works fine if I select EGA/VGA. I suspect it is the reason because any XT machine can boot with this card. 

The other issue is, even the video  drivers for Windows 1.0x works fine, The Z-Paint software for Windows exits the machine right to MS-DOS with some screen garbage. I tried enabling and disabling the NMI option on the card configuration with no avail.
I'm not sure if this is a bug or a software compatibility problem, but this software works even with the Tandy 1000 machines with Tandy 640x200 4 color video mode on Win 2.x, even if the opened image files are duplicated (This was a real behaviour and not a bug after all).

Here is the CFG and HDD image files. the HDD is a 10mb HDD can be configured as IDE or MFM with the Xebec controller.

[286 Sigma Color 400.zip](https://github.com/86Box/86Box/files/10042567/286.Sigma.Color.400.zip)
[8088 DOS 211 +Win 10.zip](https://github.com/86Box/86Box/files/10042568/8088.DOS.211.%2BWin.10.zip)

PD: I tested with with te version 3.8.0.4273, Win32, old dynarec the image files contains two copies of Win 1.04, the Windows directory is configured with the 640x400 4 color driver and Win16 have the 640x400 8 color driver.
